stdlib/internal/SwiftSyntax/Trivia.swift

```diff
@@ -0,0 +1,279 @@
+    case "Newline":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .newlines(value)
+    case "Backtick":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .backticks(value)
+    case "LineComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .lineComment(value)
+    case "BlockComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .blockComment(value)
+    case "DocLineComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .docLineComment(value)
+    case "DocBlockComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .docLineComment(value)
+    default:
+      let context =
+        DecodingError.Context(codingPath: [CodingKeys.kind],
+                              debugDescription: "invalid TriviaPiece kind \(kind)")
+      throw DecodingError.valueNotFound(String.self, context)
+    }
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    switch self {
+    case .blockComment(let comment):
+      try container.encode("BlockComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .docBlockComment(let comment):
+      try container.encode("DocBlockComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .docLineComment(let comment):
+      try container.encode("DocLineComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .lineComment(let comment):
+      try container.encode("LineComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .formfeeds(let count):
+      try container.encode("Formfeed", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .backticks(let count):
+      try container.encode("Backtick", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .newlines(let count):
+      try container.encode("Newline", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .spaces(let count):
+      try container.encode("Space", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .tabs(let count):
+      try container.encode("Tab", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .verticalTabs(let count):
+      try container.encode("VerticalTab", forKey: .kind)
+      try container.encode(count, forKey: .value)
+      
+    }
+  }
+  
+  /// A space ' ' character.
+  case spaces(Int)
+
+  /// A tab '\t' character.
+  case tabs(Int)
+
+  /// A vertical tab '\v' character.
+  case verticalTabs(Int)
+
+  /// A form-feed '\f' character.
+  case formfeeds(Int)
+
+  /// A newline '\n' character.
+  case newlines(Int)
+  
+  /// A backtick '`' character, used to escape identifiers.
+  case backticks(Int)
```

Comment:
Not sure how often you'd have more than one backtick in a row in source code, but I guess the `Int` is free in this representation :wink: